core/domain/rights_manager.py

```diff
@@ -1434,3 +1443,275 @@ def unpublicize_collection(committer_id, collection_id):
+
+
+def check_can_delete_exploration(user_id, user_actions, exploration_rights):
+    """Checks whether the user can delete given exploration.
+
+    Args:
+        user_id: str or None. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        exploration_rights: rights_object or None. Rights object of given
+            exploration.
+
+    Returns:
+        bool. Whether the user can delete given exploration.
+    """
+    if exploration_rights is None:
+        return False
+
+    if (exploration_rights.is_private() and
+            (role_services.ACTION_DELETE_OWNED_PRIVATE_EXPLORATION in (
+                user_actions)) and
+            exploration_rights.is_owner(user_id)):
+        return True
+    elif (exploration_rights.is_published() and
+          role_services.ACTION_DELETE_ANY_PUBLIC_EXPLORATION in (
+              user_actions)):
+        return True
+    return False
+
+
+def check_can_modify_exploration_roles(
+        user_id, user_actions, exploration_rights):
+    """Checks whether the user can modify roles for given exploration.
+
+    Args:
+        user_id: str or None. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        exploration_rights: rights_object or None. Rights Object of given
+            exploration.
+
+    Returns:
+        bool. Whether the user can modify roles for given exploration.
+    """
+    if exploration_rights is None:
+        return False
+
+    if (exploration_rights.community_owned or
+            exploration_rights.cloned_from):
+        return False
+
+    if (role_services.ACTION_MODIFY_ROLES_FOR_ANY_EXPLORATION in
+            user_actions):
+        return True
+    if (role_services.ACTION_MODIFY_ROLES_FOR_OWNED_EXPLORATION in
+            user_actions):
+        if exploration_rights.is_owner(user_id):
+            return True
+    return False
+
+
+def check_can_release_ownership(user_id, user_actions, exploration_rights):
+    """Checks whether the user can release ownership for given exploration.
+
+    Args:
+        user_id: str or None. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        exploration_rights: rights_object or None. Rights Object of given
+            exploration.
+
+    Returns:
+        bool. Whether the user can release ownership for given exploration.
+    """
+    if exploration_rights is None:
+        return False
+
+    if exploration_rights.is_private():
+        return False
+
+    return check_can_modify_exploration_roles(
+        user_id, user_actions, exploration_rights)
+
+
+def check_can_publish_exploration(user_id, user_actions, exploration_rights):
+    """Checks whether the user can publish given exploration.
+
+    Args:
+        user_id: str or None. Id of the user.
+        user_actions: list(str). List of actions the user can perform.
+        exploration_id: rights_object or None. Rights Object of given
```

Comment:
Argument misnamed. Please check all docstrings in this file to ensure that they are accurate.